libdnf/utils/utils.hpp

```diff
@@ -48,6 +50,16 @@ namespace filesystem {
 bool exists (const std::string &name);
 bool isDIR(const std::string& dirPath);
 std::vector<std::string> getDirContent(const std::string &dirPath);
+
+/**
+* @brief Decompress file.
+*
+* @param inPath Path to input (compressed) file
+* @param outPath Path to output (decompressed) file
+* @param outMode Mode of created (output) file
+* @param compressType Type of compression (".bz2", ".gz", ...), nullptr - detect from inPath filename. Defaults to nullptr.
```

Comment:
Does this really take strings like ".bz2", ".gz" etc. as an argument? With the leading dot? That looks like a really weird interface. This is usually done using an enumeration which contains the list of supported compression algorithms. Where is the list of supported algorithms for this?

(I'm not against the autodetection, but for an API this looks a bit loose)

Reply:
> Where is the list of supported algorithms for this?

In the libsolv? The compressType string is passed to `solv_xfopen_fd` -> libsolv function. 
Libsolv automaticaly determines compression method from input filename (from extension of the filename). And the `compressType` is injected to `solv_xfopen_fd` function instead of input filename. So `compressType` can be used to decompress files without extension.

Reply:
Ok... so my point of ".bz2", ".gz" etc. being quite loosely defined values for the compression algorithm to use still stands (this is an API function and I think this should be more specific).

If I, as a user of the function, want to know what compression algorithms are supported, what do I have to do here? Right now I have to look in the source code, find the decompression is actually done in libsolv and go there looking for the list. Doesn't seem documented there either, it's easy to find in the source, but still you need to dig deep to actually find out what are the valid values for this argument! At the very least, the documentation should state the decompression algorithms are those supported by libsolv, but I don't find it quite sufficient personally...

So an alternative would be to add an enum listing the compression algorithms (along with an AUTO value), which means maintaining a duplicate list of the algorithms supported by libsolv. Just saying...